ci_framework/roles/reproducer/tasks/push_code.yml

```diff
@@ -11,8 +11,8 @@
       ansible.builtin.git:
         dest: "{{ repo.project.src_dir }}"
         repo: "https://{{ repo.project.canonical_name }}"
-        refspec: "pull/{{ repo.change }}/head:{{ job_id }}"
-        version: "{{ job_id }}"
+        refspec: "pull/{{ repo.change }}/head:{{ job_id | default('main') }}"
+        version: "{{ job_id | default('main') }}"
```

Comment:
we may go with `omit` as well in case of no job_id, it will get the default branch HEAD